### PR TITLE
Allows non-humans to be chem scanned via the health analyzer

### LINKED ISSUE #32449: Health Analyzer chemical scan doesn't work on monkeys
Apply reagent to monkey, switch health scanner to scan chemicals and use it on monkey to find that there's nothing in the monkey. Examining the datum/reagent of the monkey shows that they have whatever reagents but the scanner won't list any.

### PR DESCRIPTION
**Pull request by vuonojenmustaturska**
Fixes #32449

[Changelogs]:

:cl: Naksu
fix: Nonhumans such as monkeys can now be scanned for chemicals with the health analyzer.
/:cl:

[why]:

### CODE REVIEW

**Comment by vuonojenmustaturska** [comment]
For the record, as always, i have actually tested this on simple animals and such

**Comment by SgtxRootbeer** [comment]
Did you test this on humans?

**Comment by vuonojenmustaturska** [comment]
humans are still mobs

File: code/game/objects/items/devices/scanners.dm [hunk 1]
```diff
@@ -275,7 +275,7 @@ MASS SPECTROMETER
 			to_chat(user, "<span class='notice'>[cyberimp_detect]</span>")
 
 /proc/chemscan(mob/living/user, mob/living/M)
-	if(ishuman(M))
+	if(ismob(M))
 		var/mob/living/carbon/human/H = M
```
Comment (RemieRichards) [inline code/game/objects/items/devices/scanners.dm]: you're playing a dangerous game

Comment (PKPenguin321) [reply code/game/objects/items/devices/scanners.dm]: i believe iscarbon is a check, should use that and cut the typecasting back to carbon

Comment (vuonojenmustaturska) [reply code/game/objects/items/devices/scanners.dm]: I initially used iscarbon, but simple_animals are not carbon and it kinda felt better if they worked too. And they do, I've actually tested them as I said. The reagents don't do much for them but it's not like the health analyzer cares about that.

Comment (RemieRichards) [reply code/game/objects/items/devices/scanners.dm]: use ```isliving()``` not ```ismob()```, living will cover simples and carbons, where as mob covers ghosts, AI camera eyes, and all sorts of DONT LOOK AT THESE mobs.

then cut the casting back to just ```var/mob/living/L = M```

**Comment by vuonojenmustaturska** [comment]
i didn't even notice the human var below, though it didn't matter because all the vars used were defined on /atom. Changed the code to be living-tier, because i didn't actually test ghosts and i'm at work.

File: code/game/objects/items/devices/scanners.dm (hunk 1 above)
Comment (vuonojenmustaturska) [reply code/game/objects/items/devices/scanners.dm]: there's no need to cast when M is supposed to be living already

Comment (RemieRichards) [reply code/game/objects/items/devices/scanners.dm]: yes, that's what i meant, and is fine.

**Pull request merged**